client/app/telecom/telephony/alias/dashboard/telecom-telephony-alias-dashboard.controller.js

```diff
@@ -0,0 +1,180 @@
+angular.module('managerApp').controller('TelecomTelephonyAliasDashboardController', class TelecomTelephonyAliasHomeController {
+  constructor(
+    $state, $stateParams, $translate, $uibModal,
+    ChartjsFactory, OvhApiTelephony, Toast,
+    voipService,
+    TELEPHONY_ALIAS_CONSUMPTION,
+  ) {
+    this.$state = $state;
+    this.$translate = $translate;
+    this.$uibModal = $uibModal;
+    this.ChartjsFactory = ChartjsFactory;
+    this.OvhApiTelephony = OvhApiTelephony;
+    this.Toast = Toast;
+    this.voipService = voipService;
+
+    this.TELEPHONY_ALIAS_CONSUMPTION = TELEPHONY_ALIAS_CONSUMPTION;
+
+    this.billingAccount = $stateParams.billingAccount;
+    this.serviceName = $stateParams.serviceName !== 'default' ? $stateParams.serviceName : null;
+  }
+
+  $onInit() {
+    this.hideDetails = false;
+    this.number = null;
+    this.consumption = {
+      incoming: {},
+      outgoing: {},
+      chart: new this.ChartjsFactory(
+        angular.copy(this.TELEPHONY_ALIAS_CONSUMPTION.chart),
+      ),
+    };
+
+    this.fetchService();
+  }
+
+  fetchService() {
+    this.loading = true;
+    this.voipService.fetchSingleService(this.billingAccount, this.serviceName).then((number) => {
+      if (number) {
+        this.number = number;
+        this.featureTypeLabel = this.$translate.instant(`telephony_alias_configuration_configuration_type_${this.number.featureType}`);
+
+        return this.voipService.getServiceDirectory(number).then((result) => {
+          this.number.directory = result.directory;
+
+          return this.hasConsumption() ? this.fetchServiceConsumption() : result;
+        });
+      }
+      return null;
+    }).catch((error) => {
+      this.Toast.error([this.$translate.instant('telephony_alias_load_error'), _.get(error, 'data.message').join(' ')]);
+    }).finally(() => {
+      this.loading = false;
+    });
+  }
+
+  fetchServiceConsumption() {
+    function extractIncomingCallsDetails(calls) {
```

Comment:
Well you don't just `extract` data, you also mutate them, and the name of the function doesn't convey this notion